config/http_config.go

```diff
@@ -146,7 +153,15 @@ func NewHTTPClientFromConfig(cfg *HTTPClientConfig) (*http.Client, error) {
 	}
 
 	if cfg.BasicAuth != nil {
-		rt = NewBasicAuthRoundTripper(cfg.BasicAuth.Username, Secret(cfg.BasicAuth.Password), rt)
+		pass := cfg.BasicAuth.Password
+		if cfg.BasicAuth.PasswordFile != "" {
+			bs, err := ioutil.ReadFile(cfg.BasicAuth.PasswordFile)
```

Comment:
This still needs to be read on every request.

Reply:
@brian-brazil The Bearer auth doesn't read it's file on each request. I'm fine with changing this, but I think both options should be consistent with each other. 

https://github.com/prometheus/common/blob/master/config/http_config.go#L133-L142

Reply:
This is due to us ending up with duplicate code over in Prometheus. They both should read on every request.

Reply:
@brian-brazil Hmm. ok. Can we break the `NewBasicAuthRoundTripper` interface? I need to pass through the passwordFile param.

Reply:
https://github.com/prometheus/prometheus/blob/master/util/httputil/client.go already implements this, you can copy that and we'll then delete that duplication later.